### ISSUE TITLE
Support event constructors

### ISSUE THREAD

**Issue by ilinsky** [clientside/amplesdk#108]
There is a trend enabling creation of events using only constructor: 

``` javascript
var oEvent = new HashChangeEvent('hashchange', {bubbles: true, cancelable: false, oldURL:"old", newURL: "new"})
```

vs. old school

``` javascript
var oEvent = document.createEvent("HashChangeEvent");
oEvent.initHashChangeEvent("hashchange", true, false, "old", "new")
```

Some developers may find it useful

**Comment by untoldone** [comment]
On a completely unrelated note -- thanks for posting this as I couldn't easily find docs for initHashChangeEvent.

**Comment by ilinsky** [comment]
Fixed per rev e577191272016d8efb5011c30a58b76adf161136 + 48f0dc4987d17308619e99eca85642b79fb1bfea
